strategies/affected/src/main/java/org/arquillian/smart/testing/strategies/affected/ClassFileIndex.java

```diff
@@ -82,96 +80,99 @@ public ClassFileIndex(ClasspathProvider classpath, List<String> globPatterns) {
                 changedTestClasses.add(javaClass);
             }
         }
         builder.clear();
-        return changedTestClasses;
     }
 
-    public JavaClass findJavaClass(String classname) {
-        JavaClass clazz = findClass(classname);
-        if (clazz == null) {
-            clazz = builder.getClass(classname);
-            if (clazz.locatedInClassFile()) {
-                addToIndex(clazz);
-            }
-        }
-        return clazz;
-    }
-
-    private JavaClass findClass(String classname) {
-        for (JavaClass jClass : graph.vertexSet()) {
-            if (jClass.getName().equals(classname)) {
-                return jClass;
-            }
-        }
-        return null;
+    private void addToIndex(JavaElement javaElement, String[] imports) {
+        addToGraph(javaElement);
+        updateJavaElementWithImportReferences(javaElement, imports);
     }
 
-    private void addToIndex(JavaClass newClass) {
-        addToGraph(newClass);
-        updateParentReferences(newClass);
-    }
-
-    private void addToGraph(JavaClass newClass) {
+    private void addToGraph(JavaElement newClass) {
         if (!graph.addVertex(newClass)) {
             replaceVertex(newClass);
         }
     }
 
-    private List<JavaClass> getParents(JavaClass childClass) {
+    private List<JavaElement> getParents(JavaElement childClass) {
         return predecessorListOf(graph, childClass);
     }
 
-    private void replaceVertex(JavaClass newClass) {
-        List<JavaClass> incomingEdges = getParents(newClass);
+    private void replaceVertex(JavaElement newClass) {
+        List<JavaElement> incomingEdges = getParents(newClass);
 
         graph.removeVertex(newClass);
         graph.addVertex(newClass);
-        for (JavaClass each : incomingEdges) {
+        for (JavaElement each : incomingEdges) {
             graph.addEdge(each, newClass);
         }
     }
 
-    private void updateParentReferences(JavaClass parentClass) {
-        for (String child : parentClass.getImports()) {
-            JavaClass childClass = findJavaClass(child);
-            if ((childClass != null) && !childClass.equals(parentClass)) {
-                if (graph.containsVertex(childClass)) {
-                    graph.addEdge(parentClass, childClass);
+    private void updateJavaElementWithImportReferences(JavaElement javaElementParentClass,
+        String[] imports) {
+        for (String importz : imports) {
+            JavaElement importClass = new JavaElement(importz);
+            if (!importClass.equals(javaElementParentClass)) {
+                if (graph.containsVertex(importClass)) {
+                    graph.addEdge(javaElementParentClass, importClass);
                 } else {
-                    graph.addVertex(childClass);
-                    graph.addEdge(parentClass, childClass);
+                    graph.addVertex(importClass);
+                    graph.addEdge(javaElementParentClass, importClass);
```

Comment:
We always `graph.addEdge(javaElementParentClass, importClass);` so maybe reverse the `if` statement to only add vertex when does not exist?